plugins/out_s3/s3.c

```diff
@@ -960,9 +984,18 @@ static int s3_put_object(struct flb_s3 *ctx, const char *tag, time_t create_time
         c = mock_s3_call("TEST_PUT_OBJECT_ERROR", "PutObject");
     }
     else {
-        c = s3_client->client_vtable->request(s3_client, FLB_HTTP_PUT,
-                                              uri, body, body_size,
-                                              NULL, 0);
+        if (ctx->canned_acl == NULL) {
+            c = s3_client->client_vtable->request(s3_client, FLB_HTTP_PUT,
+                                                  uri, body, body_size,
+                                                  NULL, 0);
+        }
+        else {
+            canned_acl_header = create_canned_acl_header(ctx->canned_acl);
```

Comment:
you have to check if `canned_acl_header` was returned as NULL or not, just in case the allocation failed. If it failed you need to log an error, clean up, and then return -1 from this function

Reply:
Got it.. I thought the error message in the `create_canned_acl_header` is enough so I didn't add it here. Will update the code